### ISSUE TITLE
addon-notes and addon-onedevice-notes out of sync switching between stories

### ISSUE THREAD

**Issue by leotm** [storybookjs/react-native#53]
**Describe the bug**

_@storybook/addon-notes_ in the browser and _@storybook/addon-ondevice-notes_ in the simulator are out of sync when switching between stories.

Notice however knobs are for example.

**To Reproduce**

Screenshare video: https://share.getcloudapp.com/z8uxRej8

**Expected behavior**
Syncronised behavior, like with _@storybook/addon-knobs_ and _@storybook/addon-ondevice-knobs_. See video.

**System:**

```bash

Environment Info:

  System:
    OS: macOS 10.15.4
    CPU: (12) x64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
  Binaries:
    Node: 13.5.0 - /usr/local/bin/node
    Yarn: 1.22.4 - ~/Documents/GitHub/Spotifire/node_modules/.bin/yarn
    npm: 6.13.4 - /usr/local/bin/npm
  Browsers:
    Safari: 13.1
  npmPackages:
    @storybook/addon-actions: ^5.3.18 => 5.3.18 
    @storybook/addon-knobs: ^5.3.18 => 5.3.18 
    @storybook/addon-links: ^5.3.18 => 5.3.18 
    @storybook/addon-notes: ^5.3.18 => 5.3.18 
    @storybook/addon-ondevice-knobs: ^5.3.18 => 5.3.18 
    @storybook/addon-ondevice-notes: ^5.3.18 => 5.3.18 
    @storybook/addons: ^5.3.18 => 5.3.18 
    @storybook/eslint-config-storybook: ^2.4.0 => 2.4.0 
    @storybook/linter-config: ^2.5.0 => 2.5.0 
    @storybook/react-native: ^5.3.18 => 5.3.18 
    @storybook/react-native-server: ^5.3.18 => 5.3.18 
  npmGlobalPackages:
    @storybook/cli: 3.4.11
```

**Additional context**

https://github.com/storybookjs/react-native/issues/24#issuecomment-629800222

**Comment by Gongreg** [comment]
Thanks. Most likely the notes addon is not listening to story change event.

**Comment by dannyhw** [comment]
Should be resolved in v6.5
